### PR TITLE
Improve type validation DX

### PR DESCRIPTION
**Pull request by delvedor**
Other than having nicer logs, this new script reduces the build time form ~40s to ~20s on my machine 🚀 

Example command:
```sh
make validate api=xpack.info type=request stack-version=8.1.0-SNAPSHOT
```

Once merged, you should run `make setup-env`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,24 +1,10 @@
 SHELL := /bin/bash
 
-validation-all:  ## Run Validation on all Endpoints
-	@echo ">> running validations for all endpoints .."
-	./run-validations.sh
+validate: ## Validate a given endpoint request or response
```
Comment (philkra) [inline Makefile:3]: is stack version mandatory? If so, I'd prefer to add a test like [here](https://github.com/elastic/elasticsearch-specification/pull/1174/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L12)

**Review by philkra — APPROVED** [review]
One question, otherwise - LGTM

File: Makefile (hunk 1 above)
Comment (delvedor) [reply Makefile:3]: Yes, it is. We **could** fallback top a default value, but then we would need to update it every minor release.

Comment (philkra) [reply Makefile:3]: Yea, that is not very convenient. Let me make a suggestion

File: Makefile [hunk 2]
```diff
@@ -1,24 +1,10 @@
...
+	@node compiler/run-validations.js --api $(api) --type $(type) --stack-version $(stack-version)
```
Comment (philkra) [inline Makefile:4]:
```suggestion
  test -n "$(stack-version)" # missing stack-version param
	@node compiler/run-validations.js --api $(api) --type $(type) --stack-version $(stack-version)
```

**Review by philkra — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (philkra) [reply Makefile:4]: downstream check makes this irrelevant

**Review by philkra — APPROVED** [review]
LGTM

File: Makefile (hunk 2 above)
Comment (delvedor) [reply Makefile:4]: The script runs the validation of the parameters.

**Review by Anaethelion — APPROVED** [review]
Kudos for speed and better logs!

LGTM

**Pull request merged**